app/models/custom_field/calculated_value.rb

```diff
@@ -39,6 +39,34 @@ module CustomField::CalculatedValue
   # Field formats that can be used within a formula.
   FIELD_FORMATS_FOR_FORMULA = %w[int float calculated_value].freeze
 
+  def self.calculator_instance
+    Dentaku::Calculator.new(case_sensitive: true)
+  end
+
+  class_methods do
+    # Select custom fields of type calculated_value that are listed in
+    # changed_cf_ids, or are referencing custom fields in changed_cf_ids either
+    # directly or through other calculated fields
+    def affected_calculated_fields(changed_cf_ids)
+      return [] if changed_cf_ids.empty?
+
+      to_check = select(&:field_format_calculated_value?)
```

Comment:
This could also filter directly on the database instead of having to first construct an instance for each row and then calling a method on it:

```suggestion
      to_check = where(field_format: "calculated_value")
```

Since this topic comes up multiple times in this review, it shows potential for a scope on the CustomField model that returns calculated values. Then we would not have to use the field format name as a string constant in situations like these 🤔

Reply:
Nice idea, went with the scope